### DISCUSSION TITLE
How to modify the position of the shape just added?

### DISCUSSION THREAD

**Discussion by AbyssShang** [pymupdf/PyMuPDF#2566] [Q&A] [answered]
I added a rectangle to the page and inserted text inside it, now I want to change it. The position or other parameters of this rectangle and text

**Comment by JorjMcKie** [comment, marked as answer]
This is not possible - with highlevel methods.
You can try and hack the `/Contents` source, which requires in-depth knowledge of the PDF appearance source syntax, see chapter "Operator Summary" on page 643 of the PDF spec.
